pkg/api/types.go

```diff
@@ -338,12 +339,41 @@ type Result struct {
 
 // BuildInfo holds information about a particular step in the build process.
 type BuildInfo struct {
+	// StepInfo holds information about build steps.
+	StepInfo []BuildStepInfo
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
 	// build, failed.
 	FailureReason FailureReason
 }
 
+// BuildStepInfo holds information about a build step.
+type BuildStepInfo struct {
+	// StepName is the name of the build step.
+	StepName StepName
+
+	// StepDuration is the duration of a build step.
+	StepDuration time.Duration
```

Comment:
i think we might want the start time and the stop time instead of just the duration.  clients can calculate the duration from the start/stop time if they want, or display the actual start/stop time if desired.